src/Factory/DbalFactory.php

```diff
@@ -54,8 +55,10 @@ public function __invoke(ContainerInterface $container): DatabaseManager
      */
     private function prepareLogger(ContainerInterface $container, mixed $logger): LoggerInterface
     {
-        if (is_string($logger)) {
-            $logger = $container->get($logger);
+        if (is_string($this->logger)) {
```

Comment:
I suggest to prepare logger in package configuration (https://github.com/yiisoft/yii-cycle/blob/master/config/di.php#L39) instead of package code.